packages/contact-center/station-login/tests/station-login/index.tsx

```diff
@@ -1,21 +1,38 @@
-import '@testing-library/jest-dom';
 import React from 'react';
-import {render, screen, cleanup} from '@testing-library/react';
-
+import {render, screen} from '@testing-library/react';
+import {StationLogin} from '../../src';
 import * as helper from '../../src/helper';
-import {StationLogin} from '../../src/station-login/index';
-
-describe('StationLogin', () => {
-  afterEach(cleanup);
+import '@testing-library/jest-dom';
 
-  it('CheckboxWithLabel changes the text after click', () => {
-    const stationLoginHelperSpy = jest.spyOn(helper, 'useStationLogin').mockReturnValue({name: 'MockStationLogin'});
+ // Sample login parameters
+ const loginReqParam = {
+  teamId: 'team123',
+  loginOption: 'EXTENSION',
+  dialNumber: '1001',
+};
+const teams = ['team123', 'team456'];
 
-    render(<StationLogin />);
+const loginOptions = ['EXTENSION', 'AGENT_DN', 'BROWSER'];
 
-    expect(stationLoginHelperSpy).toHaveBeenCalledWith();
+// Mock the store import
+jest.mock('@webex/cc-store', () => {return {
+  webex: {},
+  teams,
+  loginOptions,
+  loginReqParam,
+  setDeviceType: jest.fn(),
+  setDialNumber: jest.fn(),
+  setTeam: jest.fn()
+}});
 
+describe('StationLogin Component', () => {
+  it('renders StationLoginPresentational with correct props', () => {
+    const useStationLoginSpy = jest.spyOn(helper, 'useStationLogin');
+    
+    render(<StationLogin />);
+    console.log(useStationLoginSpy.mock.results[0].value);
+    expect(useStationLoginSpy).toHaveBeenCalledWith({webex: {}, loginReqParam});
```

Comment:
We should check number of times called. Can we create `toBeCalledOnceWith()` like in calling sdk or web client?

Reply:
toBeCalledOnceWith gives error as below and even after adding the jest macther like how we have done it in Calling SDK, issue doesn't resolve.

Property 'toBeCalledOnceWith' does not exist on type 'JestMatchers<SpyInstance<{ name: string; login: () => void; logout: () => void; loginSuccess: any; loginFailure: Error | undefined; logoutSuccess: any; }, [props: UseStationLoginProps], any>>'.

We might need to look into this separately to troubleshoot it.

Reply:
Can we create a jira to track this in the widget epic please? @Kesari3008

Reply:
Created : https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-583535